tests/AutoReview/FixerFactoryTest.php

```diff
@@ -646,6 +646,10 @@ private static function getFixersPriorityGraph(): array
                 'no_extra_blank_lines',
                 'no_spaces_inside_parenthesis',
             ],
+            'nullable_type_declaration' => [
+                'ordered_types',
+                'types_spaces',
+            ],
```

Comment:
```suggestion
            'nullable_type_declaration' => [
                'nullable_type_declaration_for_default_null_value',
                'ordered_types',
                'types_spaces',
            ],
```
This fixer must run after [`nullable_type_declaration_for_default_null_value`](https://cs.symfony.com/doc/rules/function_notation/nullable_type_declaration_for_default_null_value.html) because when `string $foo = null` is fixed to `?string $foo = null` it should be standardised - either keep `?string $foo` or change to `string|null $foo` (default value is out of scope at this point).

Reply:
I think you mean before? Because if you used the option `['use_nullable_type_declaration' => false]`, the `?` will be removed and this fixer will not know the type is nullable. I did not add priority test because this fixer's priority is higher than `nullable_type_declaration_for_default_value` fixer. If you want, I can add.

Reply:
I believe it should be _after_, regardless of `use_nullable_type_declaration` - if it's `true`, `?` will be added and then standardised to required syntax, if it's `false`, then `?` will be removed and `nullable_type_declaration` won't do anything and it's OK. Otherwise it would lead to situation where nullable type would be standardised to `Foo|null $foo = null` and `nullable_type_declaration_for_default_value` wouldn't be able to remove `null` because it does not support unions. So I think we need proper priorities OR support for unions in that fixer 🙂.

Reply:
Support for unions is in #5819

Reply:
Hah, that's great! The bad part is that it's outdated 😅. But OK, let's continue here - we can [cover it](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/5819#issuecomment-1580577776) in #5819.

Reply:
@paulbalandan #5819 was merged, but in the end it adds `?` for single types and `|null` for union/intersection types (the latter are turned into DNF). This means this fixer should be run after [`nullable_type_declaration_for_default_null_value`](https://cs.symfony.com/doc/rules/function_notation/nullable_type_declaration_for_default_null_value.html) in order to enforce consistency for nullability in type declaration.

Reply:
As far as I see in the end it was resolved differently, but it works really nice 🙂:

<img width="1651" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/99fc7f2c-058b-4913-9882-1774703b1dbf">

@paulbalandan can we add integration test like on the screen above? I mean `{"nullable_type_declaration": {"syntax": "union"}, "ordered_types": {"null_adjustment": "always_last"}, "nullable_type_declaration_for_default_null_value": true}`.